neevaco/neeva-ios
# Explicit Preview Cookie Management for Cheatsheet

We saw on mode there is a decent chunk of preview requests
It's a known issue that preview mode only works because of a WKWebkit Bug

This change introduces an explicit control of the preview cookie used for cheatsheet.
if use is not signed in, we fall back to a preview flow. In this flow, we check for a preview cookie in the keychain. If there is no cookie in the keychain, we ping neeva.com/preview to fetch a new one. Then, we use a separate GraphQLAPI instance with a separate cookie storage to make the search request.

## Checklists

### How was this tested?

### Manual test cases
on real device:
- No sign in, make cheatsheet request without visiting neeva.com first
- Then sign in, make another cheatsheet request
- Then sign out, make another cheatsheet request

- manually set an expired cookie in keychain, then make a request

Observe the header in the request and response to make sure that the cookies are correct.

### Screenshots and screen recordings

## Issues addressed
#3907
